pkg/steps/pod_test.go

```diff
@@ -182,3 +184,133 @@ func TestPodStepExecution(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPodObject(t *testing.T) {
+	testCases := []struct {
+		name        string
+		podStep     func(*podStep)
+		expectedPod func(*v1.Pod)
+	}{
+		{
+			name: "no secret name results in no mounted secrets",
+			podStep: func(expectedPodStepTemplate *podStep) {
+				expectedPodStepTemplate.config.SecretName = ""
+			},
+			expectedPod: func(expectedPodTemplate *v1.Pod) {
```

Comment:
So for instance ... right now to understand this test case I need to understand the template and then understand how the mutation is working. Instead, you can do a in-place validation with a `func(*coreapi.Pod, *testing.T)`:

```go
expectedPod: func(input *coreapi.Pod, t *testing.T) {
    if len(input.Spec.Volumes) != 0 {
        t.Errorf("declared volumes when none were expected: %v", diff.ObjectReflectDiff([]coreapi.Volume{}, input.Spec.Volumes)
    }
    if len(input.Spec.Containers[0].VolumeMounts) != 0 {
        t.Errorf("mounted volumes when none were expected: %v", diff.ObjectReflectDiff([]coreapi.VolumeMount{}, input.Spec.Containers[0].VolumeMounts)
    }
}
```


expected

Reply:
I like your minimalistic style, but at this place, we want to know if what we set is per expectation. Not only if it is set. Refactored to middle ground what was before and what are you suggesting.